### PR TITLE
movie rewiew api post/get

### PR DESCRIPTION
**Pull request by Refref18**
movie review api posta nd get methods are initalized 
example get: http://127.0.0.1:5000/movie/?keyword=life
example post: http://127.0.0.1:5000/movie/ 
example json:
 {
      "display_title": "Apocalypse '45",
      "mpaa_rating": "",
      "critics_pick": 0,
      "byline": "Natalia Winkelman",
      "headline": "‘Apocalypse ’45’ Review: Graphic Images of Wartime",
      "summary_short": "Candid testimonies from World War II veterans accompany vivid archival footage in this immersive 
     documentary that showcases the myths we tell ourselves about war.",
     "link": "https://www.nytimes.com/2021/05/27/movies/apocalypse-45-review.html"
        
      
      
 }
ps: json only requires display_title, byline and link, others are optional

### CODE REVIEW

**Review by ihsangkcl — CHANGES_REQUESTED** [review]
There are some lines with absolute path. They can cause problems in different computers. I think these should be relative paths.

**Comment by veyisTurgut** [comment]
> There are some lines with absolute path. They can cause problems in different computers. I think these should be relative paths.

Mr. Gökcül, sqlite sometimes confuse with relative paths. It can stay as full path for now, we can convert those in deployment phase.

**Comment by veyisTurgut** [comment]
> movie review api posta nd get methods are initalized
> example get: http://127.0.0.1:5000/movie/?keyword=life
> example post: http://127.0.0.1:5000/movie/
> example json:
> {
> "display_title": "Apocalypse '45",
> "mpaa_rating": "",
> "critics_pick": 0,
> "byline": "Natalia Winkelman",
> "headline": "‘Apocalypse ’45’ Review: Graphic Images of Wartime",
> "summary_short": "Candid testimonies from World War II veterans accompany vivid archival footage in this immersive
> documentary that showcases the myths we tell ourselves about war.",
> "link": "https://www.nytimes.com/2021/05/27/movies/apocalypse-45-review.html"
> 
> }
> ps: json only requires display_title, byline and link, others are optional

You better put this documentation to [here](https://github.com/bounswe/2021SpringGroup12/wiki/Practice-app-Documentation)

File: practice-app/mapper.py [hunk 1]
```diff
@@ -1,14 +1,46 @@
...
+        ) else "",
+        byline=movie["byline"],
+        headline=movie["headline"] if 'headline' in movie.keys() else "",
+        summary_short=movie["summary_short"] if 'summary_short' in movie.keys(
+        ) else "",
+        link=movie["link"]["url"]
+    )
+
 
+def movie_mapper2(movie: dict) -> Movie:
```
Comment (veyisTurgut) [inline practice-app/mapper.py:34]: Why there are 2 different functions for same purpose, can you explain?

File: practice-app/practice-app.py [hunk 2]
```diff
@@ -32,8 +33,9 @@ def get_books():
     if "name" not in request.args:
         return Response("Please provide an author name!", status=400)
     # now we are sure name parameter is supplied, request books of this author from NYTimes API.
-    author_name = request.args.get("name").title().replace(" ","+")
-    r = requests.get("https://api.nytimes.com/svc/books/v3/reviews.json?author={}&api-key=Ug3r4KRJxcd69bNw5i89CWxArqiGlrGB".format(author_name))
+    author_name = request.args.get("name").title().replace(" ", "+")
```
Comment (veyisTurgut) [inline practice-app/practice-app.py:36]: Try not to change others' lines please.

File: practice-app/practice-app.py [hunk 3]
```diff
@@ -82,32 +88,98 @@ def create_book():
...
+            continue
+    con.commit()
+    con.close()
 
+    return schemas.MovieResponse(movies=movies).__dict__
 
+# ---------------------------------------------------------------------------------------
 
 
+@app.route('/movie/', methods=['POST'])
```
Comment (veyisTurgut) [inline practice-app/practice-app.py:145]: Its a best practice to make endpoint paths plural. I mean: /movie -> /movies

File: practice-app/sqlfiles/create_tables.sql [hunk 4]
```diff
@@ -19,3 +19,17 @@ CREATE TABLE IF NOT EXISTS BookISBNs (
...
+    mpaa_rating         TEXT,
+    critics_pick        INTEGER,
+    headline            TEXT,
+    summary_short       TEXT,
+    link                TEXT NOT NULL,
+	PRIMARY KEY(movie_id)
+);
+
+ 
+ Movie(display_title, mpaa_rating, critics_pick, byline, headline,summary_short, link)
```
Comment (veyisTurgut) [inline practice-app/sqlfiles/create_tables.sql:35]: Should this line be included in .sql file?

**Review by veyisTurgut — COMMENTED** [review]
I wrote comments, please check them out and reply.

**Comment by ihsangkcl** [comment]
> > There are some lines with absolute path. They can cause problems in different computers. I think these should be relative paths.
> 
> Mr. Gökcül, sqlite sometimes confuse with relative paths. It can stay as full path for now, we can convert those in deployment phase.

OK. I dismissed the change request. 👍

**Comment by veyisTurgut** [comment]
> > > There are some lines with absolute path. They can cause problems in different computers. I think these should be relative paths.
> > 
> > 
> > Mr. Gökcül, sqlite sometimes confuse with relative paths. It can stay as full path for now, we can convert those in deployment phase.
> 
> OK. I dismissed the change request.

Just came into my mind, we can make something like:
`import OS
path = OS.getcwd()
sqlite3.connect(path+"/.sqlfiles/..")`

File: practice-app/practice-app.py (hunk 3 above)
Comment (yamanogluberk) [reply practice-app/practice-app.py:145]: Agreed

File: practice-app/mapper.py (hunk 1 above)
Comment (Refref18) [reply practice-app/mapper.py:34]: one directly gets the link from user :dict[link] (movie_mapper2) other gets the link from the url which is a key in link dictionary: dic[link][url] (movie_mapper)

Comment (veyisTurgut) [reply practice-app/mapper.py:34]: Ah! I see. Still, even though  I'm pretty sure that you can manage to combine these two into one function., you can leave it like this.

File: practice-app/sqlfiles/create_tables.sql (hunk 4 above)
Comment (Refref18) [reply practice-app/sqlfiles/create_tables.sql:35]: forget to delete it sorrrrrrrrrrry

**Comment by veyisTurgut** [comment]
Any updates? @Refref18

**Comment by veyisTurgut** [comment]
You just opened this: https://github.com/bounswe/2021SpringGroup12/pull/180. 
I think this pr can be closed

**Comment by batuhanovski** [comment]
I wanted to note that, before we decided to change the front-end form 'HTML' to 'vue', we arranged a little meeting with Refika and reviewed each others code, gave each other feedbacks about 'HTML' form of the front-end.

**Pull request closed without merging**